### ISSUE TITLE
Maybe swizzle layoutSubviews, setNeedsLayout, and sizeThatFits: to avoid subclassing

### ISSUE THREAD

**Issue by johnboiles** [YOLayout/YOLayout#16]
Feedback I got from a fellow dev was that he's super hesitant to subclass UIView. I think I understand; I think I'd feel the same if I hadn't been a part of creating YOLayout.

Opening this issue to discuss the possibility of swizzling in the required layout/sizing methods (probably `setFrame:`, `layoutSubviews`, `sizeThatFits:`, `setNeedsLayout`) and implementing layout as a category of UIView. Benefit of this is that it should work on `UIControl`, `UIButton`, `UITableViewCell` & `UIView` right out of the box.

I think we'd still keep YOView subclass around --the `viewInit` method itself is useful enough to merit this subclass. But we'd also provide the option of adding a YOLayout to an arbitrary UIView subclass.

What do you think?

**Comment by johnboiles** [comment]
Maybe the category would be something like this. In this method all the necessary swizzling would happen as well as the creation and storage of the `YOLayout` object.

```Objective-C
@interface UIView (YOLayout)
- (void)yo_setLayoutBlock:(YOLayoutBlock)layoutBlock
@end
```

**Comment by johnboiles** [comment]
Cool thing about this, is that it _should_ just work with UIControl, UIScrollView, etc

**Comment by johnboiles** [comment]
Closed by #53
